sensor/kubernetes/certrefresh/utils_test.go

```diff
@@ -0,0 +1,138 @@
+package certrefresh
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+var (
+	backoff = wait.Backoff{
+		Duration: 100 * time.Millisecond,
+		Factor:   1.2,
+		Jitter:   0.1,
+		Steps:    3,
+	}
+	sensorPodName   = "sensor-pod"
+	sensorNamespace = "stackrox-ns"
+	replicaSetName  = "sensor-replica-set"
+	deploymentName  = "sensor-deployment"
+)
+
+func createTestPod(podName, namespace, ownerName string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "apps/v1",
+					Kind:       "ReplicaSet",
+					Name:       ownerName,
+					UID:        "replica-set-uid",
+				},
+			},
+		},
+	}
+}
+
+func createTestReplicaSet(rsName, namespace, ownerName string) *appsv1.ReplicaSet {
+	return &appsv1.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rsName,
+			Namespace: namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+					Name:       ownerName,
+					UID:        "deployment-uid",
+				},
+			},
+		},
+	}
+}
+
+func TestFetchSensorDeploymentOwnerRef_Success(t *testing.T) {
```

Comment:
All functions below this line can be aggregated into single table-driven test function. That would make it easier to read and add testcases in the future.

Reply:
Done